09_json/willshen8/cmd/puppy-server/main.go

```diff
@@ -0,0 +1,49 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+
+	"github.com/anz-bank/go-course/09_json/willshen8/pkg/puppy"
+	jsonify "github.com/anz-bank/go-course/09_json/willshen8/pkg/puppy/json"
+	store "github.com/anz-bank/go-course/09_json/willshen8/pkg/puppy/store"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+var out io.Writer = os.Stdout
+
+var (
+	args     = os.Args[1:]
+	fileName = kingpin.Flag("data", "Puppy Json File Name").Short('d').String()
+)
+
+func main() {
+	processCommandLineArgs()
+}
+
+func processCommandLineArgs() {
+	var jsonData []byte
+	var decodedPuppies []puppy.Puppy
+	var mystore = store.NewMapStore()
+
+	_, parseError := kingpin.CommandLine.Parse(args)
+	if parseError != nil {
+		fmt.Fprintln(out, "Incorrect arguments! Please use flag --help for usage.")
+	}
+
+	jsonData, error := ioutil.ReadFile(*fileName)
+
+	if error != nil {
+		panic(error)
+	}
+
+	jsonify.UnmarshalPuppy(jsonData, &decodedPuppies)
+
+	fmt.Fprintln(out, decodedPuppies)
+	for _, jsonPuppy := range decodedPuppies {
+		jsonPuppy := jsonPuppy //this fix lint issue
```

Comment:
what linter issue? I'm surprised this is a problem as you are not starting a go routine?

Reply:
thanks for clarifying.

Reply:
👍

Reply:
should probably be `// fixes linter issue`